### PR TITLE
fix(link-with-icon): add size lg and fix masthead styles

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#6179 #6212 

### Description

update link with icon to use large size from Carbon. update masthead close icon styles. Use carbon's link paired with icon in megamenu.

### Changelog

**New**

- large size for link

**Changed**

- clean up close icon styles

### CODE REVIEW

File: packages/web-components/tests/snapshots/dds-text-cta.md [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 ```
 <a
-  class="bx--link bx--link-with-icon bx--link-with-icon__icon-right"
+  class="bx--link bx--link--lg bx--link-with-icon bx--link-with-icon__icon-right"
```
Comment (photodow) [inline packages/web-components/tests/snapshots/dds-text-cta.md:27]: Is there a reason we need to touch so many components to make this change? Are we not depending on link with icon in these higher up components?

**Review by photodow — COMMENTED** [review]
I was expecting the opportunity to remove some CSS in this PR. Is there a reason we wouldn't be removing something like at least type tokens in this PR?

File: packages/web-components/tests/snapshots/dds-text-cta.md (hunk 1 above)
Comment (ariellalgilmore) [reply packages/web-components/tests/snapshots/dds-text-cta.md:27]: We are depending on link with icon. These are the snapshot files that automatically get updated in web components. The only thing I touched was adding the large size in Link with Icon, which is now being showed everywhere link with icon is used.

**Review by photodow — CHANGES_REQUESTED** [review]
Looks like the font size increased in the mega menu. I don't think that was desired?

image

File: packages/web-components/tests/snapshots/dds-text-cta.md (hunk 1 above)
Comment (photodow) [reply packages/web-components/tests/snapshots/dds-text-cta.md:27]: I see now. I saw snapshot in a couple of the file names, but missed it in some of the others it appears. Thanks for clarification!

**Comment by oliviaflory** [comment]
> Looks like the font size increased in the mega menu. I don't think that was desired?
> 
> image


😢 I hadn't gotten to looking at mega menu yet. Does that mean we need two different link with icons? One for 14px and one for 16px type pairing?

**Comment by ariellalgilmore** [comment]
> OR (re-thinking the above) would we utilize the Carbon link with icon variant in the mega menu to align with the productive nature of the mega menu?

I was thinking the same thing @oliviaflory. The only thing is Carbon web components does not have that variant yet, so we would have to include that first

**Comment by oliviaflory** [comment]
> > OR (re-thinking the above) would we utilize the Carbon link with icon variant in the mega menu to align with the productive nature of the mega menu?
> 
> I was thinking the same thing @oliviaflory. The only thing is Carbon web components does not have that variant yet, so we would have to include that first

Makes sense @ariellalgilmore

Would you recommend: 
1. Waiting to approve this PR, and opening another issue to a) add the link with icon to Carbon web components and b) fix the mega menu styling ? 
2. Approving this PR and opening another issue to a) add the link with icon to Carbon web components and b) fix the mega menu styling ? 

My understanding is this work is all within a feature branch and won't affect the current build of mega menu, so I'm ok with either approach.

**Comment by ariellalgilmore** [comment]
> > > OR (re-thinking the above) would we utilize the Carbon link with icon variant in the mega menu to align with the productive nature of the mega menu?
> > 
> > 
> > I was thinking the same thing @oliviaflory. The only thing is Carbon web components does not have that variant yet, so we would have to include that first
> 
> Makes sense @ariellalgilmore
> 
> Would you recommend:
> 
> 1. Waiting to approve this PR, and opening another issue to a) add the link with icon to Carbon web components and b) fix the mega menu styling ?
> 2. Approving this PR and opening another issue to a) add the link with icon to Carbon web components and b) fix the mega menu styling ?
> 
> My understanding is this work is all within a feature branch and won't affect the current build of mega menu, so I'm ok with either approach.

I think let's go with option a. Hopefully it won't take too long to get the link with icon in Carbon web components. I will turn this PR into a draft for now.

**Review by kennylam — APPROVED** [review]
LGTM thanks @ariellalgilmore!

**Comment by oliviaflory** [comment]
@ariellalgilmore This looks great! A few comments:

### 1. Web components mega menu category link styling 
The title has changed to blue and lost the arrow icon within all menus
Screen Shot 2021-06-23 at 11 34 46 AM

Expected: 
Screen Shot 2021-06-23 at 11 36 48 AM

### 2. Web components mega menu links lost the icon
The links lost the icon within the "featured" section of the Products & Solutions menu
Screen Shot 2021-06-23 at 12 34 36 PM

Expected:
Screen Shot 2021-06-23 at 12 35 03 PM


### 3. Web components; the View all products and View all Services buttons lost their icon
Screen Shot 2021-06-23 at 12 36 26 PM

Expected:
Screen Shot 2021-06-23 at 12 36 51 PM

_**Note: React looks fine, so I'm not sure where the styling is coming from that Web components is different.**_

Let me know if you want to meet to discuss the above, I'm available!

**Review by kennylam — APPROVED** [review]

**Review by oliviaflory — APPROVED** [review]
woo hoo looks great @ariellalgilmore !

**Review by annawen1 — APPROVED** [review]
LGTM!

**Pull request merged**